scripts/merge-i18n-files.ts

```diff
@@ -0,0 +1,100 @@
+ * * Source directory: The directory with override files
+ *   - Required
+ *   - Recommended to place override files in the theme directory under assets/i18n (but this is not required)
+ *   - Files must have matching names in both source and destination directories, for example:
+ *        en.json5 in the source directory will be merged with en.json5 in the destination directory
+ *        fr.json5 in the source directory will be merged with fr.json5 in the destination directory
+ */
+function parseCliInput() {
+  program
+    .option('-d, --output-dir <output-dir>', 'output dir when running script on all language files', projectRoot(LANGUAGE_FILES_LOCATION))
+    .option('-s, --source-dir <source-dir>', 'source dir of transalations to be merged')
+    .usage('(-s <source-dir> [-d <output-dir>])')
+    .parse(process.argv);
+
+  if (program.outputDir && program.sourceDir) {
+    if (!fs.existsSync(program.outputDir) && !fs.lstatSync(program.outputDir).isDirectory() ) {
+      console.error('Output does not exist or is not a directory.');
+      console.log(program.outputHelp());
+      process.exit(1);
+    }
+    if (!fs.existsSync(program.sourceDir) && !fs.lstatSync(program.sourceDir).isDirectory() ) {
+      console.error('Source does not exist or is not a directory.');
+      console.log(program.outputHelp());
+      process.exit(1);
+    }
+    fs.readdirSync(projectRoot(program.sourceDir)).forEach(file => {
+      if (fs.existsSync(program.outputDir + '/' + file) )
+      {
+        console.log("Merging: "+ program.outputDir + '/' + file + ' with '+ program.sourceDir + '/' + file)
+        mergeFileWithSource(program.sourceDir + '/' + file, program.outputDir + '/' + file)
+      }
+    });
+  } else {
+    console.error('Source or Output parameter is missing.');
+    console.log(program.outputHelp());
+    process.exit(1);
+  }
+}
+
+/**
+ * Reads source file and output file to merge the contents
+ *  > Iterates over the source file keys
+ *  > Updates values for each key and adds new keys as needed
+ *  > Updates the output file with the new merged json
+ * @param pathToSourceFile Valid path to source file to merge from
+ * @param pathToOutputFile Valid path to merge and write output
+ */
+function mergeFileWithSource(pathToSourceFile, pathToOutputFile) {
+  const progressBar = new _cliProgress.SingleBar({}, _cliProgress.Presets.shades_classic);
+  progressBar.start(100, 0);
+
+  const source_file = fs.readFileSync(pathToSourceFile, 'utf8')
+  progressBar.update(10);
+  const output_file = fs.readFileSync(pathToOutputFile, 'utf8')
+  progressBar.update(20);
+
+  let parsed_source = JSON5.parse(source_file)
+  progressBar.update(30);
+  let parsed_output = JSON5.parse(output_file)
+  progressBar.update(40);
+
+  for (let key of Object.keys(parsed_source)) {
+    parsed_output[key] = parsed_source[key];
+  }
+  progressBar.update(80);
+  fs.writeFileSync(pathToOutputFile,JSON5.stringify(parsed_output,{space:'\n  '}), {encoding:'utf8'})
```

Comment:
If you add the `quote` option here, the output file will keep using double quotes

```suggestion
  fs.writeFileSync(pathToOutputFile,JSON5.stringify(parsed_output,{ space:'\n  ', quote: '"' }), { encoding:'utf8' });
```

Reply:
Thanks Art, the result with double quotes still pulls out duplicate newlines and unnecessary spaces, but it's certainly easier to see the changes.